### PR TITLE
Jl apikeys

### PR DESCRIPTION
**Pull request by jlacefie**
## What does this PR do?
Added release stage callout for both API Keys and Service Accounts docs
Refactored the API Keys docs for readability 
Tested and fixed the CLI and SDK examples

### CODE REVIEW

File: docs/production-deployment/cloud/api-keys.mdx [hunk 1]
```diff
@@ -21,64 +21,59 @@ tags:
...
 
 :::
 
 Temporal Cloud supports secure programmatic access through API Key Authentication.
 
 ## Overview
 
-Temporal Cloud API Keys facilitate user-level authentication.
-Each key is unique to a single user and ties directly to their Role-Based Access Control (RBAC) settings, ensuring secure and appropriate access.
+Temporal Cloud API Keys provide identity, User and [Service Account](/cloud/service-accounts), based authentication.
```
Comment (mastermanu) [inline docs/production-deployment/cloud/api-keys.mdx:33]: what does it mean to say identity, User and Service Account authentication?

File: docs/production-deployment/cloud/api-keys.mdx [hunk 2]
```diff
@@ -87,15 +82,24 @@ You must enable API Key access to allow the creation of API Keys.
...
 
 1. [Login](https://cloud.temporal.io/) to the Cloud UI.
 2. [Select **Settings** and choose **API Keys**](https://cloud.temporal.io/settings/api-keys).
 
-From here, you can update access to the API Key for an account using the **Enable** and **Disable** toggle.
-You can also disable or delete an individual User API Key using the vertical ellipsis at the right of the API Keys row.
+You can also disable or delete an individual API Key using the vertical ellipsis at the right of the API Keys row. Disabling API Keys disables the creation of API Keys. Existing API Keys can be used to authenticate into Temporal Cloud until they are either disabled, deleted or expire.
+
+:::note
+Deleting a key removes the ability for that API Key to authenticate into Temporal Cloud. Be aware that if you delete an API Key for a running WF, that WF will fail until a new API Key secret is created and configured.
```
Comment (mastermanu) [inline docs/production-deployment/cloud/api-keys.mdx:95]: if you delete a key for a "running worker/client".

File: docs/production-deployment/cloud/api-keys.mdx [hunk 3]
```diff
@@ -87,15 +82,24 @@ You must enable API Key access to allow the creation of API Keys.
...
+Global Administrators can monitor, manage, disable, and delete API Keys for any User or Service Account within their account.
 
 To manage your accounts API Keys:
 
 1. [Login](https://cloud.temporal.io/) to the Cloud UI.
 2. [Select **Settings** and choose **API Keys**](https://cloud.temporal.io/settings/api-keys).
 
-From here, you can update access to the API Key for an account using the **Enable** and **Disable** toggle.
-You can also disable or delete an individual User API Key using the vertical ellipsis at the right of the API Keys row.
+You can also disable or delete an individual API Key using the vertical ellipsis at the right of the API Keys row. Disabling API Keys disables the creation of API Keys. Existing API Keys can be used to authenticate into Temporal Cloud until they are either disabled, deleted or expire.
```
Comment (mastermanu) [inline docs/production-deployment/cloud/api-keys.mdx:92]: Feels second part of this row "Disabling API Keys..." should be on its own line

File: docs/production-deployment/cloud/api-keys.mdx [hunk 4]
```diff
@@ -21,64 +21,59 @@ tags:
...
 
-These API Keys are integral for authenticating Temporal Cloud operator tools, including [Temporal Cloud CLI (tcld)](/cloud/tcld) and the [Temporal Cloud Operations API](/ops).
+API Keys support 2 main use cases:
 
-Users have the autonomy to create, delete, and update access to their API Keys through the Cloud UI or tcld.
+- _**Cloud Operations Automation**_: API Keys can be used as the authentication mechanism for Temporal Cloud operational tools ([Temporal Cloud CLI (tcld)](/cloud/tcld), [the Cloud Ops API](/ops), [Terraform](/production-deployment/cloud/terraform-provider)) to manage your Temporal Cloud account, namespaces, certificates, and user identities.
+  - API Keys for use in Cloud Operations Tools are in Public Preview
+- _**Namespace Authentication**_: API Keys can be used as the Namespace authentication mechanism for executing and managing workflows from the SDK and Temporal CLI as an alternative to using mTLS-based authentication.
+  - API Keys for use in Namespace Authentication are in Pre Release
+  - If you created a Temporal Account using the Self Serve Signup feature, API Keys are available for Namespace Authentication.
```
Comment (mastermanu) [inline docs/production-deployment/cloud/api-keys.mdx:48]: How does a customer know/remember how they signed-up? IIRC, there is no indicator once the Account is created that would tell them this

File: docs/production-deployment/cloud/api-keys.mdx (hunk 1 above)
Comment (jlacefie) [reply docs/production-deployment/cloud/api-keys.mdx:33]: IIUC, a User and a Service Account are Identity Types.  
We use Identities in the product at https://cloud.temporal.io/settings/identities 
The Identity language in the docs aligns to the language in the product.

File: docs/production-deployment/cloud/api-keys.mdx (hunk 2 above)
Comment (jlacefie) [reply docs/production-deployment/cloud/api-keys.mdx:95]: ack - making this change

File: docs/production-deployment/cloud/api-keys.mdx (hunk 3 above)
Comment (jlacefie) [reply docs/production-deployment/cloud/api-keys.mdx:92]: ack - making this change

File: docs/production-deployment/cloud/api-keys.mdx (hunk 4 above)
Comment (jlacefie) [reply docs/production-deployment/cloud/api-keys.mdx:48]: removing these sections per slack convo

File: docs/production-deployment/cloud/api-keys.mdx (hunk 1 above)
Comment (mastermanu) [reply docs/production-deployment/cloud/api-keys.mdx:33]: I think there is a colon missing here because the sentence as constructed suggests there are 3 distinct items where as there is really Identity: User and Service Account

Comment (jlacefie) [reply docs/production-deployment/cloud/api-keys.mdx:32]: ack - will make this more clear

Comment (jlacefie) [reply docs/production-deployment/cloud/api-keys.mdx]: language changed. marking this one done

**Review by jsundai — COMMENTED** [review]
First set of review.

File: docs/production-deployment/cloud/api-keys.mdx [hunk 5]
```diff
@@ -21,64 +21,59 @@ tags:
 
 :::tip Support, stability, and dependency info
 
-- In [Public Preview](/evaluate/development-production-features/release-stages#public-preview)
+- API Keys used for Cloud Operations Tools (tcld, API, Terraform) are In [Public Preview](/evaluate/development-production-features/release-stages#public-preview)
```
[the remark this answers is not in the document]
Comment (jlacefie) [reply docs/production-deployment/cloud/api-keys.mdx]: fixed

File: docs/production-deployment/cloud/api-keys.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jlacefie) [reply docs/production-deployment/cloud/api-keys.mdx]: fixed

File: docs/production-deployment/cloud/service-accounts.mdx [hunk 6]
```diff
@@ -79,6 +87,7 @@ While User identities are invited to Temporal Cloud, Service Accounts are create
 5. (Optional) Create API Key
 
 - It is recommended to create an API Key for the Service Account right after you create the Service Account, though you can create/manage API Keys for Service Accounts at any time.
+- see the API Key [documentation](/cloud/api-keys) for more information on creating and managing API Keys
```
[the remark this answers is not in the document]
Comment (jlacefie) [reply docs/production-deployment/cloud/service-accounts.mdx]: fixed

File: docs/production-deployment/cloud/api-keys.mdx [hunk 7]
```diff
@@ -21,64 +21,59 @@ tags:
...
+- API Keys used for Namespace Authentication are In [Pre Release](/evaluate/development-production-features/release-stages#pre-release)
```
[the remark this answers is not in the document]
Comment (jlacefie) [reply docs/production-deployment/cloud/api-keys.mdx]: fixed

File: docs/production-deployment/cloud/api-keys.mdx [hunk 8]
```diff
@@ -21,64 +21,56 @@ tags:
...
+- API Keys used for Namespace Authentication are in [Pre Release](/evaluate/development-production-features/release-stages#pre-release)
+  :::
 
 Temporal Cloud supports secure programmatic access through API Key Authentication.
 
 ## Overview
 
-Temporal Cloud API Keys facilitate user-level authentication.
-Each key is unique to a single user and ties directly to their Role-Based Access Control (RBAC) settings, ensuring secure and appropriate access.
+Temporal Cloud API Keys provide identity based authentication where an Identity can be a Temporal User or [Service Account](/cloud/service-accounts)
```
Comment (jsundai) [inline docs/production-deployment/cloud/api-keys.mdx:32]:
```suggestion
Temporal Cloud API Keys provide identity-based authentication where an Identity can be either a Temporal User or a [Service Account](/cloud/service-accounts).
```

File: docs/production-deployment/cloud/api-keys.mdx [hunk 9]
```diff
@@ -21,64 +21,56 @@ tags:
...
+- [Temporalʼs Terraform Provider](/production-deployment/cloud/terraform-provider)
 
-To use your API Key with the [Cloud Ops API](/ops), securely pass your API Key in your Cloud Ops API client.
+## Get started {#get-started}
 
-For a complete example, see [Cloud Samples in Go](https://github.com/temporalio/cloud-samples-go/blob/1dd4254b6ed1937e361005c0144410e72b8a5542/client/api/apikey.go).
+### Prerequisites
 
-## Manage API Keys {#manage-api-keys}
+- Global Adinistrator should [Enable access to API Keys](#manage-api-keys)
```
Comment (jsundai) [inline docs/production-deployment/cloud/api-keys.mdx:67]:
```suggestion
- Global Administrator should [Enable access to API Keys](#manage-api-keys)
```

File: docs/production-deployment/cloud/api-keys.mdx [hunk 10]
```diff
@@ -122,11 +127,11 @@ To generate an API Key using the Cloud UI:
 
 :::note
 
-Temporal supports up to a 90-day duration for API Keys.
+Temporal currently supports up to a 90-day duration for API Keys. The expiration will be extended in a future release.
```
Comment (jsundai) [inline docs/production-deployment/cloud/api-keys.mdx:130]: Do we need to include this? Fine with either way though. 
"The expiration will be extended in a future release."

File: docs/production-deployment/cloud/api-keys.mdx [hunk 11]
```diff
@@ -142,13 +147,13 @@ tcld apikey create --name Your-Key-Name --description YourDescription --duration
...
 
-Use tcld to create API Keys with a day level expiration by day.
+Use tcld to create API Keys with a day level expiration.
 
 :::
 
 ### Delete an API Key
 
-Delete an API Key using one of the following methods.
+Delete an API Key using the Cloud UI or tcld. Deleting a key removes the ability for that API Key to authenticate into Temporal Cloud. Be aware that if you delete an API Key for a running WF, that WF will fail unless you rotate the API Key for a new one.
```
Comment (jsundai) [inline docs/production-deployment/cloud/api-keys.mdx:156]: I would change the abbreivation here too!

File: docs/production-deployment/cloud/api-keys.mdx [hunk 12]
```diff
@@ -21,64 +21,56 @@ tags:
...
+  - API Keys for use in Namespace Authentication are in Pre Release
 
-API Keys can be used in the `tcld` or in the Cloud Ops API.
+API Keys are an opt-in feature that must be enabled by a Global Administrator for use. For instructions on API Key Management for your organization, see [Global Administrator API Key Management](#ga-manage-api-keys).
 
-**How can I use the API Key with tcld?**
+API Keys are supported for Users and Service Accounts.
 
-To use your API Key with tcld, use one of the following methods:
+- Users have the autonomy to create, delete, and update access to their API Keys through the Cloud UI or tcld. For broader oversight, Global Administrators can manage user's API Keys using the same interfaces.
```
Comment (jsundai) [inline docs/production-deployment/cloud/api-keys.mdx:52]: I would take away the period on these bullet points so it's consistent with the rest.

File: docs/production-deployment/cloud/api-keys.mdx [hunk 13]
```diff
@@ -197,55 +200,102 @@ tcld apikey disable --id <api-key-id>
...
+### Create an API Key for a Service Account
+
+Create an API Key for a Service Account using the Temporal Cloud UI or tcld.
+
+#### Using the Cloud UI
 
-### Overview
+1. Navigate to https://cloud.temporal.io/settings/api-keys
+2. Click the Create API Key button
+3. Click Service Account at the top of the page under "Create an API Key for*"
```
Comment (jsundai) [inline docs/production-deployment/cloud/api-keys.mdx:217]:
```suggestion
3. Click Service Account at the top of the page under "Create an API Key for"
```

Comment (jsundai) [reply docs/production-deployment/cloud/api-keys.mdx:217]: Is there a * here or maybe a typo?

File: docs/production-deployment/cloud/api-keys.mdx [hunk 14]
```diff
@@ -197,55 +200,102 @@ tcld apikey disable --id <api-key-id>
...
+- Expiration date (required - defaulted to 30 days)
+
+:::note
+
+Temporal currently supports up to a 90-day duration for API Keys.
+
+:::
+
+6. Click the Generate API Key button
+7. You will see the new API keyʼs secret, if the API Key was created successfully.
```
Comment (jsundai) [inline docs/production-deployment/cloud/api-keys.mdx:233]:
```suggestion 
7. You will see the new API Keyʼs secret, if the API Key was created successfully.
```

**Review by jsundai — COMMENTED** [review]
Second set of review.

File: docs/production-deployment/cloud/api-keys.mdx (hunk 8 above)
Comment (fairlydurable) [reply docs/production-deployment/cloud/api-keys.mdx:32]: Identity is not listed as one of our uppercase terms. If so, please file a ticket so I can update our linter checks (probably as a warning) or consider lower-case.

Comma before where

File: docs/production-deployment/cloud/api-keys.mdx [hunk 15]
```diff
@@ -490,77 +533,22 @@ myClient.Connection.RpcMetadata = new Dictionary<string, string>()
...
-
-The process for enabling/disabling an API Key is the same for a User or a Service Account.
+To use an API Key with tcld, use one of the following methods:
 
-### Delete an API Key for a Service Account
+- `--api-key` flag
+- `TEMPORAL_CLOUD_API_KEY` environment variable
 
-Global Administrators have ability to delete individual API Keys for any user in their account using the Cloud UI or tcld.
+### The Cloud Ops API
```
Comment (jsundai) [inline docs/production-deployment/cloud/api-keys.mdx:543]:
```suggestion
### Cloud Ops API
```

File: docs/production-deployment/cloud/api-keys.mdx [hunk 16]
```diff
@@ -490,77 +533,22 @@ myClient.Connection.RpcMetadata = new Dictionary<string, string>()
...
+To use an API Key with the [Cloud Ops API](/ops), securely pass your API Key in your Cloud Ops API client.
 
-1. Navigate to https://cloud.temporal.io/settings/identities
-2. Find the identity that owns the API Key to disable/enable and click on the row to see the API Keys associated with the identity.
-3. Click the Delete button. There may be a delay after changing the status of an API Key.
-4. Once successful, the API Keys status will be reflected in the row.
+For a complete example, see [Cloud Samples in Go](https://github.com/temporalio/cloud-samples-go/blob/1dd4254b6ed1937e361005c0144410e72b8a5542/client/api/apikey.go).
 
-#### Using tcld
+### The Terraform Provider
```
Comment (jsundai) [inline docs/production-deployment/cloud/api-keys.mdx:549]:
```suggestion 
### Terraform Provider
```

File: docs/production-deployment/cloud/api-keys.mdx (hunk 9 above)
Comment (fairlydurable) [reply docs/production-deployment/cloud/api-keys.mdx:67]: This is missing a pronoun before Global.

Same issue with Global Administrator. De-cap or add to linter as official capped term.

Make sure "Enable access to API keys" is either fixed for capitalization or adjust wording to indicate that this is verbatim so it can be found on a web page, normally by "selecting 'Enable access to API Keys' option on the Web interface" or similar.

File: docs/production-deployment/cloud/api-keys.mdx (hunk 10 above)
Comment (fairlydurable) [reply docs/production-deployment/cloud/api-keys.mdx:130]: IMO it's best to never mention future changes in docs. We may do it. We may not. Make no promises. Similarly avoid "currently". The docs show what is accurate to this feature.

Same question/ticketing about capitalization for API Keys. 

Suggested rewording:

Temporal supports API keys lifespans up to 90 days. Your key will be valid during this period unless it is revoked or the account is deactivated.

File: docs/production-deployment/cloud/api-keys.mdx (hunk 11 above)
Comment (fairlydurable) [reply docs/production-deployment/cloud/api-keys.mdx:156]: We have not yet resolved a peer consensus about one sentence per line on docs. This practice allows cleaner diffs and feedback for revision. Happy to go along with whatever the team decides on.

Don't use WF. We use full words like Workflow.

Suggested rewording: 

Deleting an API key deactivates its permissions to authenticate into Temporal Cloud.
Please note that this affects running Workflows that were initialized with a deleted API key except when the API key was rotated into place.
This can affect long-running Workflows.

And then add a link to the API key rotation process in-text.

File: docs/production-deployment/cloud/api-keys.mdx (hunk 12 above)
Comment (fairlydurable) [reply docs/production-deployment/cloud/api-keys.mdx:52]: "Autonomy" is a $5 word.

Users may create, delete, and update API key access using the Cloud Web UI or the `tcld` CLI tool.
Global Administrators have permission to manage user API keys using the same interfaces.

I'm using "Cloud Web UI" per previous tickets. Please check with team cloud to confirm.

File: docs/production-deployment/cloud/api-keys.mdx (hunk 13 above)
Comment (fairlydurable) [reply docs/production-deployment/cloud/api-keys.mdx:217]: If quotes are used here, they should be used consistently through this list and the document. When wording is taken from UIs, make sure they match exactly including punctuation such as colons and ellipses.

File: docs/production-deployment/cloud/api-keys.mdx (hunk 14 above)
Comment (fairlydurable) [reply docs/production-deployment/cloud/api-keys.mdx:233]: See earlier comments about capitalization.

This is not a step as the user does nothing here. You can make it a step by adding "Confirm". I think, however, the action will be something along the lines of "Confirm that the new API key's secret token has been successfully generated and copy it to a safe place." followed by "The key will not be shown in full form at any other point" or something along those lines (please check for accuracy).

File: docs/production-deployment/cloud/api-keys.mdx (hunk 16 above)
Comment (fairlydurable) [reply docs/production-deployment/cloud/api-keys.mdx:549]: Provider is lower-cased throughout our site, e.g.

> ./docs/production-deployment/cloud/terraform-provider.mdx:The GitHub repository for the Terraform provider is [terraform-provider-temporalcloud](https://github.com/temporalio/terraform-provider-temporalcloud/tree/main), where you can find the source code, contribution guidelines, and additional information.

**Comment by fairlydurable** [comment]
Please note, after doing a backup pass over our feedback, I'm going to do a content review now to evaluate information structuring, explanations, etc.

File: docs/production-deployment/cloud/api-keys.mdx [hunk 17]
```diff
@@ -21,64 +21,56 @@ tags:
 
 :::tip Support, stability, and dependency info
 
-- In [Public Preview](/evaluate/development-production-features/release-stages#public-preview)
-
-:::
+- API Keys used for Cloud Operations Tools (tcld, API, Terraform) are in [Public Preview](/evaluate/development-production-features/release-stages#public-preview)
```
Comment (fairlydurable) [inline docs/production-deployment/cloud/api-keys.mdx:24]: This is the overall structure of your document.

```
## Overview
## Get started {#get-started}
### Prerequisites
## API Key Management and Governance {#manage-api-keys}
### Global Administrator API Key Management {#ga-manage-api-keys}
## User API Keys {#user-api-keys}
### Generate an API Key
#### Using the Cloud UI
#### Using tcld
### Delete an API Key
#### Using the Cloud UI
#### Using tcld
### Enable/Disable an API Key
#### Using the Cloud UI
#### Using tcld
## Managing API Keys for Service Accounts {#serviceaccount-api-keys}
### Create an API Key for a Service Account
#### Using the Cloud UI
#### Using tcld
### Enable/Disable an API Key
#### Using the Cloud UI
#### Using tcld
### Delete an API Key for a Service Account
#### Using the Cloud UI
#### Using tcld
## API Keys for Namespace Authentication {#namespace-authentication}
## Using API Keys {#using-apikeys}
### Temporal CLI
### SDK
#### Go SDK (pre v 1.26.0)
#### Java SDK
#### Python SDK (v 1.6.0+)
#### Python SDK (pre v 1.6.0)
#### .NET SDK (v 1.1.0+)
### tcld
### The Cloud Ops API
### The Terraform Provider
## Best practices {#best-practices}
## Troubleshooting {#troubleshooting}
## Frequently asked questions {#faqs}
```

From the style guide:

> Use H3 headers sparingly, and avoid H4 headers. If you need to use subheaders, make sure there are two or more headers of that level within that section of the tutorial. Alternatively, consider making multiple steps.

There are several places that do not follow the balance rule for "at least two" H3 or lower headings in this doc.

File: docs/production-deployment/cloud/api-keys.mdx [hunk 18]
```diff
@@ -21,64 +21,56 @@ tags:
...
-
-### Prerequisites
-
-- A Cloud user account
-- Access to the [Temporal Cloud UI](https://cloud.temporal.io/) or Temporal Cloud CLI ([tcld](https://docs.temporal.io/cloud/tcld/))
-- [Enable access to API Keys](#manage-api-keys)
+API Keys support 2 main use cases:
 
-## Use API Keys {#use-api-keys}
+- _**Cloud Operations Automation**_: API Keys can be used as the authentication mechanism for Temporal Cloud operational tools ([Temporal Cloud CLI (tcld)](/cloud/tcld), [the Cloud Ops API](/ops), [Terraform](/production-deployment/cloud/terraform-provider)) to manage your Temporal Cloud account, namespaces, certificates, and user identities.
```
Comment (fairlydurable) [inline docs/production-deployment/cloud/api-keys.mdx:43]: It's simpler to say "Use API keys to authenticate with Temporal Cloud operational tools".

Please use casing consistently. For example, we have rules about Namespace being upper-cased. Applying the Vale linter will yelp. It is part of the docs repo.

File: docs/production-deployment/cloud/api-keys.mdx (hunk 11 above)
Comment (jsundai) [reply docs/production-deployment/cloud/api-keys.mdx:156]: Good catch. Agreed that this doc should have one sentence per line. Otherwise it might end up being a  correction afterwards.

File: docs/production-deployment/cloud/api-keys.mdx [hunk 19]
```diff
@@ -21,64 +21,56 @@ tags:
...
+  - API Keys for use in Cloud Operations Tools are in Public Preview
+- _**Namespace Authentication**_: API Keys can be used as the Namespace authentication mechanism for executing and managing workflows from the SDK and Temporal CLI as an alternative to using mTLS-based authentication.
```
Comment (fairlydurable) [inline docs/production-deployment/cloud/api-keys.mdx:45]: Do we have a page to link to for mTLS-based authentication methods on-site?

File: docs/production-deployment/cloud/api-keys.mdx [hunk 20]
```diff
@@ -197,55 +200,102 @@ tcld apikey disable --id <api-key-id>
 tcld apikey enable --id <api-key-id>
 ```
 
-## API Keys for Client Connections
+## Managing API Keys for Service Accounts {#serviceaccount-api-keys}
 
-Temporal Cloud supports secure programmatic access through API Key Authentication.
+Global Administrators have the ability to manage API Keys for any Service Account in their Temporal Cloud Account.
+The process to manage API Keys for Service Accounts is very similar to the process of managing API Keys for Users with one difference: Unlike User API Keys, Global Admins are able to generate API Keys for Service Accounts.
+Disabling/enabling, deleting, and viewing API Keys for Service Accounts follows the same process as API Keys for Users.
```
Comment (fairlydurable) [inline docs/production-deployment/cloud/api-keys.mdx:207]: These two lines (206 and 207) have parallelism between the process for Users.  Does knowing that the process is similar to "user" roles really help here?

File: docs/production-deployment/cloud/api-keys.mdx [hunk 21]
```diff
@@ -269,13 +318,13 @@ Here's an example of connecting to Temporal Cloud from the CLI using an environm
...
 --namespace <namespace.accountid> \
---tls --tls-disable-host-verification \
 --grpc-meta "temporal-namespace=<namespace.accountid>"
+--tls
 ```
 
 ### SDK
 
-To use your API Key with a Temporal SDK, use headers to specify a Bearer token is provided for authentication and the Namespace for request routing.
+To use your API Key with a Temporal SDK, use headers to specify a Bearer token and a Namespace. The Bearer token header is used to indicate the authentication type of the request and the Namespace header is used for request routing.
```
Comment (fairlydurable) [inline docs/production-deployment/cloud/api-keys.mdx:327]: Is there a more specific way to refer to "headers"?   Something like "When using API keys, add your Bearer token to the Authorization header to authenticate your Temporal Cloud requests." And then something about setting the Namespace header for request routing?

Comment (fairlydurable) [reply docs/production-deployment/cloud/api-keys.mdx:327]: Also, earlier in the doc (not in the changes), you refer to KMS. Can you specify that's a key management system?

**Comment by fairlydurable** [comment]
Because it's really hard to follow with all the back and forth, I looked at the doc in the "view the file", which makes it impossible to leave in-line feedback, especially where there weren't edits. Some feedback I put into notes:

* Audit for passive voice and over-use of "you can" auxiliary wording. Vale will pick up some of this.
* can I have multiple - Can I issue and use multiple API keys for the same account?
* How many API keys can be issued at once?
* Do API keys expire?
* s/lost/lose (What happens if I misplace or lose my API bearer token/secret key?)
* In Overview, 4 paras in a row start with "API Keys". Consider making a bullet list or rewording.
* A global administrator must... (versus should, in order to activate API key use)
* Global administrators have the ability -> can. It's shorter and friendlier. Audit for "have the ability"
* The process to manage -> Manage
* tcld: You can use tcld with the `--api-key` flag or by setting the `TEMPORAL_CLOUD_API_KEY` environment variable. The environment variable will be automatically picked up by the tcld CLI utility. (I think it gets the point across better)
* Keep it secret|Keep it safe - not a big fan of the pipe character. Comma? Semicolon? lower-case the second K?
* Rotate KEYS regularly (missing word)
* Monitor API key usage (ditto)

I apologize for the brevity and my lack of time to do a thorough review. I hope this helps.
